From 0ea231cc1808477fc97966d3ac618744712fbfcc Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 22 Mar 2014 11:52:12 -0500
Subject: [PATCH 456/456] rtl8192du: Change !!memcmp() to memcmp()

The  conversion from rtw_memcmp() to memcmp() involved an inversion of
the logic. In some cases, the conversion resulted in !!memcmp().
These are now fixed..

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_mlme_ext.c     | 20 ++++++++++----------
 core/rtw_recv.c         | 19 +++++++++----------
 os_dep/ioctl_cfg80211.c |  2 +-
 3 files changed, 20 insertions(+), 21 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
+++ b/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
@@ -543,8 +543,8 @@ static void _mgt_dispatcher(struct rtw_a
 
 	if (ptable->func) {
 		/* receive the frames that ra(a1) is my address or ra(a1) is bc address. */
-		if (!!memcmp(GetAddr1Ptr(pframe), myid(&adapt->eeprompriv), ETH_ALEN) &&
-		    !!memcmp(GetAddr1Ptr(pframe), bc_addr, ETH_ALEN))
+		if (memcmp(GetAddr1Ptr(pframe), myid(&adapt->eeprompriv), ETH_ALEN) &&
+		    memcmp(GetAddr1Ptr(pframe), bc_addr, ETH_ALEN))
 			return;
 		ptable->func(adapt, precv_frame);
 	}
@@ -574,8 +574,8 @@ void mgt_dispatcher(struct rtw_adapter *
 	}
 
 	/* receive the frames that ra(a1) is my address or ra(a1) is bc address. */
-	if (!!memcmp(GetAddr1Ptr(pframe), myid(&adapt->eeprompriv), ETH_ALEN) &&
-	    !!memcmp(GetAddr1Ptr(pframe), bc_addr, ETH_ALEN)) {
+	if (memcmp(GetAddr1Ptr(pframe), myid(&adapt->eeprompriv), ETH_ALEN) &&
+	    memcmp(GetAddr1Ptr(pframe), bc_addr, ETH_ALEN)) {
 		return;
 	}
 
@@ -1157,7 +1157,7 @@ unsigned int OnAuthClient(struct rtw_ada
 	DBG_8192D("%s\n", __func__);
 
 	/* check A1 matches or not */
-	if (!!memcmp
+	if (memcmp
 	    (myid(&(adapt->eeprompriv)), get_da(pframe), ETH_ALEN))
 		return _SUCCESS;
 
@@ -1335,7 +1335,7 @@ unsigned int OnAssocReq(struct rtw_adapt
 		status = _STATS_FAILURE_;
 	else {
 		/*  check if ssid match */
-		if (!!memcmp
+		if (memcmp
 		    ((void *)(p + 2), cur->Ssid.Ssid, cur->Ssid.SsidLength))
 			status = _STATS_FAILURE_;
 
@@ -1753,7 +1753,7 @@ unsigned int OnAssocRsp(struct rtw_adapt
 	DBG_8192D("%s\n", __func__);
 
 	/* check A1 matches or not */
-	if (!!memcmp
+	if (memcmp
 	    (myid(&(adapt->eeprompriv)), get_da(pframe), ETH_ALEN))
 		return _SUCCESS;
 
@@ -2070,7 +2070,7 @@ unsigned int OnAction_back(struct rtw_ad
 	struct sta_priv *pstapriv = &adapt->stapriv;
 
 	/* check RA matches or not */
-	if (!!memcmp(myid(&(adapt->eeprompriv)), GetAddr1Ptr(pframe), ETH_ALEN))	/* for if1, sta/ap mode */
+	if (memcmp(myid(&(adapt->eeprompriv)), GetAddr1Ptr(pframe), ETH_ALEN))	/* for if1, sta/ap mode */
 		return _SUCCESS;
 
 	DBG_8192D("%s\n", __func__);
@@ -5326,7 +5326,7 @@ unsigned int on_action_public(struct rtw
 	u8 category, action;
 
 	/* check RA matches or not */
-	if (!!memcmp
+	if (memcmp
 	    (myid(&(adapt->eeprompriv)), GetAddr1Ptr(pframe), ETH_ALEN))
 		goto exit;
 
@@ -5373,7 +5373,7 @@ unsigned int OnAction_p2p(struct rtw_ada
 	DBG_8192D("%s\n", __func__);
 
 	/* check RA matches or not */
-	if (!!memcmp(myid(&(adapt->eeprompriv)), GetAddr1Ptr(pframe), ETH_ALEN))	/* for if1, sta/ap mode */
+	if (memcmp(myid(&(adapt->eeprompriv)), GetAddr1Ptr(pframe), ETH_ALEN))	/* for if1, sta/ap mode */
 		return _SUCCESS;
 
 	frame_body =
--- a/drivers/staging/rtl8192du/core/rtw_recv.c
+++ b/drivers/staging/rtl8192du/core/rtw_recv.c
@@ -860,14 +860,14 @@ static int sta2sta_data_frame(struct rtw
 			goto exit;
 		}
 
-		if ((!!memcmp(myhwaddr, pattrib->dst, ETH_ALEN)) && (!bmcast)) {
+		if ((memcmp(myhwaddr, pattrib->dst, ETH_ALEN)) && (!bmcast)) {
 			ret = _FAIL;
 			goto exit;
 		}
 
 		if (!memcmp(pattrib->bssid, "\x0\x0\x0\x0\x0\x0", ETH_ALEN) ||
 		    !memcmp(mybssid, "\x0\x0\x0\x0\x0\x0", ETH_ALEN) ||
-		    (!!memcmp(pattrib->bssid, mybssid, ETH_ALEN))) {
+		    (memcmp(pattrib->bssid, mybssid, ETH_ALEN))) {
 			ret = _FAIL;
 			goto exit;
 		}
@@ -876,7 +876,7 @@ static int sta2sta_data_frame(struct rtw
 
 	} else if (check_fwstate(pmlmepriv, WIFI_STATION_STATE) == true) {
 		/*  For Station mode, sa and bssid should always be BSSID, and DA is my mac-address */
-		if (!!memcmp
+		if (memcmp
 		    (pattrib->bssid, pattrib->src, ETH_ALEN)) {
 			RT_TRACE(_module_rtl871x_recv_c_, _drv_err_,
 				 ("bssid != TA under STATION_MODE; drop pkt\n"));
@@ -895,7 +895,7 @@ static int sta2sta_data_frame(struct rtw
 		} else {	/*  not mc-frame */
 
 			/*  For AP mode, if DA is non-MCAST, then it must be BSSID, and bssid == BSSID */
-			if (!!memcmp
+			if (memcmp
 			    (pattrib->bssid, pattrib->dst, ETH_ALEN)) {
 				ret = _FAIL;
 				goto exit;
@@ -957,7 +957,7 @@ static int ap2sta_data_frame(struct rtw_
 		}
 
 		/*  da should be for me */
-		if ((!!memcmp(myhwaddr, pattrib->dst, ETH_ALEN)) && (!bmcast)) {
+		if ((memcmp(myhwaddr, pattrib->dst, ETH_ALEN)) && (!bmcast)) {
 			RT_TRACE(_module_rtl871x_recv_c_, _drv_info_,
 				 (" ap2sta_data_frame:  compare DA fail; DA=%pM\n",
 				  pattrib->dst));
@@ -968,7 +968,7 @@ static int ap2sta_data_frame(struct rtw_
 		/*  check BSSID */
 		if (!memcmp(pattrib->bssid, "\x0\x0\x0\x0\x0\x0", ETH_ALEN) ||
 		    !memcmp(mybssid, "\x0\x0\x0\x0\x0\x0", ETH_ALEN) ||
-		    (!!memcmp(pattrib->bssid, mybssid, ETH_ALEN))) {
+		    (memcmp(pattrib->bssid, mybssid, ETH_ALEN))) {
 			RT_TRACE(_module_rtl871x_recv_c_, _drv_info_,
 				 (" ap2sta_data_frame:  compare BSSID fail ; BSSID=%pM\n",
 				  pattrib->bssid));
@@ -1061,7 +1061,7 @@ static int sta2ap_data_frame(struct rtw_
 
 	if (check_fwstate(pmlmepriv, WIFI_AP_STATE) == true) {
 		/* For AP mode, RA=BSSID, TX=STA(SRC_ADDR), A3=DST_ADDR */
-		if (!!memcmp(pattrib->bssid, mybssid, ETH_ALEN)) {
+		if (memcmp(pattrib->bssid, mybssid, ETH_ALEN)) {
 			ret = _FAIL;
 			goto exit;
 		}
@@ -1095,7 +1095,7 @@ static int sta2ap_data_frame(struct rtw_
 		}
 	} else {
 		u8 *myhwaddr = myid(&adapter->eeprompriv);
-		if (!!memcmp(pattrib->ra, myhwaddr, ETH_ALEN)) {
+		if (memcmp(pattrib->ra, myhwaddr, ETH_ALEN)) {
 			ret = RTW_RX_HANDLED;
 			goto exit;
 		}
@@ -1128,8 +1128,7 @@ static int validate_recv_ctrl_frame(stru
 	}
 
 	/* receive the frames that ra(a1) is my address */
-	if (!!memcmp
-	    (GetAddr1Ptr(pframe), myid(&padapter->eeprompriv), ETH_ALEN))
+	if (memcmp(GetAddr1Ptr(pframe), myid(&padapter->eeprompriv), ETH_ALEN))
 		return _FAIL;
 
 	/* only handle ps-poll */
--- a/drivers/staging/rtl8192du/os_dep/ioctl_cfg80211.c
+++ b/drivers/staging/rtl8192du/os_dep/ioctl_cfg80211.c
@@ -2059,7 +2059,7 @@ static int cfg80211_rtw_connect(struct w
 
 		if (sme->ssid && sme->ssid_len) {
 			if (pnetwork->network.Ssid.SsidLength != sme->ssid_len ||
-			    !!memcmp(pnetwork->network.Ssid.Ssid, (void *)sme->ssid, sme->ssid_len))
+			    memcmp(pnetwork->network.Ssid.Ssid, (void *)sme->ssid, sme->ssid_len))
 				continue;
 		}
 
